mne/epochs.py

```diff
@@ -1385,8 +1393,17 @@ def _get_data(self, out=True, picks=None, item=None, *, units=None,
         else:
             # we start out with an empty array, allocate only if necessary
             data = np.empty((0, len(self.info['ch_names']), len(self.times)))
-            logger.info('Loading data for %s events and %s original time '
-                        'points ...' % (n_events, len(self._raw_times)))
+            msg = (f'for {n_events} events and {len(self._raw_times)} '
+                   'original time points')
+            if self._decim > 1:
+                msg += ' (prior to decimation)'
+            _msg_preload = False
+            if hasattr(self._raw, "preload"):
+                _msg_preload = self._raw.preload
+            if _msg_preload:
+                logger.info(f'Using data from preloaded Raw {msg} ...')
+            else:
+                logger.info(f'Loading data {msg} ...')
```

Comment:
```suggestion
            if self._raw.preload:
                logger.info(f'Using data from preloaded Raw {msg} ...')
            else:
                logger.info(f'Loading data {msg} ...')
```

the preload attr is part of the public API of raw. Example for a raw array:

In [4]: raw2 = mne.io.RawArray(info=raw.info, data=raw.get_data())

In [5]: raw2.preload
Out[5]: True

Reply:
I had this in an earlier commit, but `self._raw` is not always `mne.io.Raw` --> it can also be `None`, or a list of `_RawContainer`, neither of which have a `preload` attribute.

Reply:
indeed...